org.eclipse.xtext/src/org/eclipse/xtext/xtext/AnnotationNames.java

```diff
@@ -13,12 +13,19 @@
 
 /**
  * @author Christian Schneider - Initial contribution and API
+ * @author Holger Schill
  */
 public final class AnnotationNames {
 
 	public static final String OVERRIDE = "Override";
 	
-	public static final List<String> VALID_ANNOTATIONS_NAMES = ImmutableList.of(OVERRIDE);
+	public static final String EXPORTED = "Exported";
+	
+	public static final String FINAL = "Final";
+	
+	public static final String DEPRECATED = "Deprecated";
+	
+	public static final List<String> VALID_ANNOTATIONS_NAMES = ImmutableList.of(OVERRIDE, EXPORTED, FINAL, DEPRECATED);
```

Comment:
Some annotations do not make sense on terminals. We need to add a validation.

Reply:
Should also be refactored to an enum.

Reply:
Validation done.

Since we need the Literals and Java does not support that out of the box - what's the benefit of refactor that to an Enum?

Reply:
We could provide some metadata in the enums, e.g. 'supportedOnTerminalRule' or 'supportedOnEnumRule'.

Reply:
Will refactor after build with tests included are green. The complete Fragment stuff is not really build to be tested with other together. I hope I found the problem now.